lib/presentation/contact_form/contact_form_screen.dart

```diff
@@ -42,134 +60,126 @@ class ContactFormState extends State<ContactFormPage> {
     _pageScrollController = ScrollController();
   }
 
+  void _contactFormBlocListener(BuildContext context, ContactFormState state) {
+    // Remove any old snackbar
+    ScaffoldMessenger.of(context).removeCurrentSnackBar();
+    state.map(
+        initial: (_) {},
+        submitting: (_) {
+          ScaffoldMessenger.of(context).showSnackBar(

```

Comment:
We should consider implementing a loading dialog that uses the WillPop scope so that we can avoid using snackbar for implementations, since it's used in a wrong way compared to material guidelines here. _(It's fine for testing, but we should make a task for it!)_

Reply:
Agreed. I didn't touch any of that, since there is no design atm.

Reply:
I'll try to get Niklas to make a design and let's push a task since if we're already implementing BLOCs we might as well do this.